### PR TITLE
Improve performance test framework

### PR DESCRIPTION
**Pull request by gyorb**
- improve performance test framework
- add new profiler options to CodeChecker
- do some source cleanup

### CODE REVIEW

**Comment by bruntib** [comment]
The `number_of_runs` option in the config file is not taken into account. The performance test generates as many runs as the number of cores used (given by `-j` flag). Moreover it seems that around 10 runs set, only runs/cores are created, but none of them contain any reports.

**Comment by Xazax-hun** [comment]
There is a conflict.

File: codechecker_lib/profiler.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import pstats
+import StringIO
+
+from codechecker_lib import logger
+
+LOG = logger.get_new_logger('PROFILER')
+
+
+def timeit(function):
+    '''
```
Comment (Xazax-hun) [inline codechecker_lib/profiler.py]: Use """ for docstrings.

File: tests/performance_test/readme.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+`bug_length`: bug path length for each bug  
+`clean_after_fill`: remove all run results after each measurement  
+`wait_for_store`: if multiple processes are used for measurement wait until the database is filled up with dummy results before starting the measurements  
+`check_port`: the check port used in the CodeChecker server command  
+`view_port`: the view port used in the CodeChecker server command  
+
+A test configuration cloud look like this:  
+```
+{
+"number_of_runs" : 1,
```
Comment (Xazax-hun) [inline tests/performance_test/readme.md]: What about adding some indentation?

**Pull request merged**